application/src/main/java/net/corda/v5/application/interop/parameters/ParameterTypeLabel.java

```diff
@@ -0,0 +1,47 @@
+package net.corda.v5.application.interop.parameters;
+
+import org.jetbrains.annotations.NotNull;
+import java.math.BigDecimal;
+import java.nio.ByteBuffer;
+import java.time.ZonedDateTime;
+import java.util.Arrays;
+
+public enum ParameterTypeLabel {
+    BOOLEAN("boolean", Boolean.class),
```

Comment:
As discussed `typeName` field could be redundant for `ParameterTypeLabel`, as each enum has an "implied/synthetic"  field `name`.In our case `typeName` has the same litaral value as enum literals (apart of upper-lower cases). Hence I would add  at least TODO to check if we can drop `typeName`.